transformer_engine/pytorch/attention.py

```diff
@@ -3106,11 +3106,18 @@ def remove_extra_states_check(self, incompatible_keys): # pylint: disable=unused
                     incompatible_keys.missing_keys.remove(key)
         self.register_load_state_dict_post_hook(remove_extra_states_check)
 
-    def get_fp8_weights_scratchpad(
-        self,
-        is_first_microbatch: Union[bool, None],
-    ) -> List[Float8Tensor]:
-        """Needs override."""
+    def _save_to_state_dict(self, destination, prefix, keep_vars):
+        """
+        Override to save to DotProductAttention's _extra_state.
+        """
+        super()._save_to_state_dict(destination, prefix.replace('fused_attention.',''), keep_vars)
+
+    def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
+        missing_keys, unexpected_keys, error_msgs):
+        """
+        Override to load nothing as DotProductAttention's _load_from_state_dict()
+        has loaded its _extra_state.
+        """
```

Comment:
I don't think it would fail when checkpointing, but I think it might have a bit of problem loading - if users have just `FusedAttention` as their model, no `DotProductAttention` on top. I guess the question is do we want to support both `core_attention._extra_state` and `core_attention.fused_attention._extra_state` in the future, or just the former, because `DotProductAttention` is our public API and `FusedAttention` is not.